### PR TITLE
display custom error details for injected web3

### LINKED ISSUE #3765: Custom error not working correctly.
Hello Members!

Custom Error is not working same as in VM and Testnet(Sepolia, Goerli..etc).
The caller will not get the reason behind why the function call is failing.

Following snippet works as expected in VM but when deployed in Sepolia...Does not provide which custom error was triggered...
```


// SPDX-License-Identifier: GPL-3.0
pragma solidity ^0.8.19;

contract Test {

    error InvalidAddressError();
    error InvalidAddressWithParamError(address _addr);
    
    function checkCustom() public pure returns (bool){
        address _input = address(0); 
        if (_input == address(0)) revert InvalidAddressError();
        return true;
    }

    function checkRequire() public pure returns (bool){
        address _input = address(1);

        require(_input == address(0), "############ Address must be non zero");
        return true;
    }

    function checkCustomParametrised() public pure returns (bool){
        address _input = address(4);

        if (_input != address(0)) revert InvalidAddressWithParamError(_input);
        return true;
    }
   
}
```

### PR DESCRIPTION
**Pull request by Aniket-Engg**
fixes #3765

### CODE REVIEW

File: libs/remix-lib/src/execution/txExecution.ts [hunk 1]
```diff
@@ -173,3 +173,77 @@ export function checkVMError (execResult, compiledContracts) {
   ret.message = `${error}\n${exceptionError}\n${msg}\nDebug the transaction to get more information.`
   return ret
 }
+
+export function checkInjectedError (errorObj, compiledContracts) {
```
[the remark this answers is not in the document]
Comment (Aniket-Engg) [reply libs/remix-lib/src/execution/txExecution.ts]: done. Please check

File: libs/remix-lib/src/execution/txExecution.ts [hunk 2]
```diff
@@ -75,20 +75,20 @@ export function checkVMError (execResult, compiledContracts) {
     error: false,
     message: ''
   }
-  if (!execResult.exceptionError) {
+  if (!execResult.exceptionError && !isInjectedWeb3) {
```
Comment (yann300) [inline libs/remix-lib/src/execution/txExecution.ts:78]: seems like `error` and `returnData` are both the same but present under different properties... could you add a type `execResult`, so you don't have to do stuff like `isInjectedWeb3 ? execResult.error : exceptionError`

Comment (yann300) [reply libs/remix-lib/src/execution/txExecution.ts:78]: that way you may not need to have the `isInjected`

Comment (Aniket-Engg) [reply libs/remix-lib/src/execution/txExecution.ts:80]: I think `execResult` is quite different for VM and injected Web3. For Injected web3, it is just an object having error details

Comment (Aniket-Engg) [reply libs/remix-lib/src/execution/txExecution.ts]: updated

**Review by yann300 — APPROVED** [review]

**Pull request merged**